compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1534,18 +1534,45 @@ impl<'a> Parser<'a> {
         Ok((ident, ItemKind::MacroDef(ast::MacroDef { body, macro_rules: false })))
     }
 
-    /// Is this unambiguously the start of a `macro_rules! foo` item definition?
-    fn is_macro_rules_item(&mut self) -> bool {
-        self.check_keyword(kw::MacroRules)
-            && self.look_ahead(1, |t| *t == token::Not)
-            && self.look_ahead(2, |t| t.is_ident())
+    /// Is this a possibly malformed start of a `macro_rules! foo` item definition?
+    ///
+    /// The first bool is whether it's a macro_rules item.
+    /// The second is whether the `!` after `macro_rules` is present
+    fn is_macro_rules_item(&mut self) -> (bool, bool) {
+        if self.check_keyword(kw::MacroRules) {
+            let macro_rules_span = self.look_ahead(0, |t| t.span);
```

Comment:
Isn't this equivalent to `self.token.span`?

Reply:
Yep, didn't realise you could do that